packages/dds/tree/src/edit-manager/editManager.ts

```diff
@@ -137,12 +135,32 @@ export class EditManager<TChangeset, TChangeFamily extends ChangeFamily<any, TCh
         return netChange;
     }
 
-    private rebaseBranch(branch: Branch<TChangeset>, newRef: SeqNumber) {
-        const trunkChanges = this.getCommitsBetween(branch.refSeq, newRef);
+    /**
+     * Updates the `branch` to reflect the local changes that the session owner would have had after
+     * they learned of the commit with sequence number `newRef` being sequenced.
+     * Concretely, this means two things:
+     * 1. Rebasing the changes in the branch over any new trunk changes up to and including `newRef`.
+     * 2. Removing from the branch those changes that were sequenced before or at sequence number `newRef`.
+     * We can purge those because any new changes from this session (such as the one whose ref is `newRef`)
+     * will be independent from them (they will instead depend the trunk version of them, or some later changes
+     * in the trunk).
```

Comment:
I tried to reword this for clarity.

```suggestion
     * This is accomplished by rebasing the branch's changes over any new trunk changes up to and including `newRef`.
     * Changes with sequence number less than or equal to `newRef` are removed from the branch,
     * since they are now part of the trunk this branch is based on.
```